contract/kernel/kernel.go

```diff
@@ -398,6 +430,32 @@ func (k *Kernel) rollbackUpdateMaxBlockSize(desc *contract.TxDesc) error {
 	return err
 }
 
+func (k *Kernel) runUpdateReservedContract(desc *contract.TxDesc) error {
+	if k.context == nil || k.context.LedgerObj == nil {
+		return fmt.Errorf("failed to update reservered contract, because no ledger object in context")
+	}
+	params, vErr := k.validateUpdateReservedContract(desc)
+	if vErr != nil {
+		return vErr
+	}
+	k.log.Info("update reservered contract")
+	err := k.context.LedgerObj.UpdateReservedContract(params)
+	return err
+}
+
+func (k *Kernel) rollbackUpdateReservedContract(desc *contract.TxDesc) error {
```

Comment:
I found runUpdateReservedContract and rollbackUpdateReservedContract are 100% same. It should be bug there.

Reply:
done